dfa/src/main/scala/dfa/DFA.scala

```diff
@@ -1,3 +1,23 @@
 package dfa // leave this line in the file
 
 // TODO: replace this comment with your implementation
+
+case class State (label:String)
+case class Transition (from:State, to:State, symbol:Char)
+
+class DFA(states:Set[State], val transitions:Set[Transition],
+val start: State, val accept:Set[State]):
+    
+  def accepts(input:String): Boolean = {
```

Comment:
I like how you approached the problem in a more functional program style with case matching. I used a for loop to take inputs one at a time, which might save some memory, but your solution is easier to read when I try to map it to how I understand DFAs from a diagram.